### PR TITLE
Some Ui Related Update

### PR DESCRIPTION
**Pull request by Ritika8081**
- [x] Added button in footer that will toggle between grid/list and change display of  canvas .
- [x] Updated autoscale button.
- [x] Added space in infinity popover
- [x] Added power button that will be able to pause and resume data graph in all canvas.

### CODE REVIEW

File: src/components/Canvas.tsx [hunk 1]
```diff
@@ -17,25 +17,28 @@ interface CanvasProps {
   data: string;
   selectedBits: BitSelection;
   isGridView: boolean;
+  isDisplay: boolean; // New prop for play/pause functionality
```
Comment (akadeepesh) [inline src/components/Canvas.tsx:20]: if there is not a good reason for using isDisplay as variable name for pause functionality we should stick to isAllPaused, it's more direct and understandable.

File: src/components/Canvas.tsx [hunk 2]
```diff
@@ -17,25 +17,28 @@ interface CanvasProps {
...
+  isDisplay,
+}) => {
+  const { theme } = useTheme();
+  const channels = useMemo(() => [true, true, true, true], []);
+  const chartRef = useRef<SmoothieChart[]>([]);
+  const seriesRef = useRef<(TimeSeries | null)[]>([]);
+  const [isChartInitialized, setIsChartInitialized] = useState(false);
+  const [isPaused, setIsPaused] = useState(Array(channels.length).fill(false)); // Paused state for each channe
+  const batchSize = 10;
+  const batchBuffer = useMemo<Array<{ time: number; values: number[] }>>(
```
Comment (akadeepesh) [inline src/components/Canvas.tsx:36]: Please don't remove added comments.

File: src/components/Canvas.tsx [hunk 3]
```diff
@@ -133,12 +131,11 @@ const Canvas: React.FC<CanvasProps> = ({ data, selectedBits, isGridView }) => {
...
 
   const processBatch = useCallback(() => {
-    // Process the batch data
-    if (batchBuffer.length === 0) return;
+    if (batchBuffer.length === 0 || !isDisplay) return;
 
     batchBuffer.forEach((batch) => {
       channels.forEach((channel, index) => {
-        if (channel && !isPaused[index]) {
+        if (channel) {
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: Can you explain reason to remove !isPaused[index] from condition check? It was making sure that data won't append to a stream which is paused.

File: src/components/Canvas.tsx [hunk 4]
```diff
@@ -17,25 +17,28 @@ interface CanvasProps {
...
+  selectedBits,
+  isGridView,
+  isDisplay,
+}) => {
+  const { theme } = useTheme();
+  const channels = useMemo(() => [true, true, true, true], []);
+  const chartRef = useRef<SmoothieChart[]>([]);
+  const seriesRef = useRef<(TimeSeries | null)[]>([]);
+  const [isChartInitialized, setIsChartInitialized] = useState(false);
+  const [isPaused, setIsPaused] = useState(Array(channels.length).fill(false)); // Paused state for each channe
```
Comment (akadeepesh) [inline src/components/Canvas.tsx:34]: Just fill this array with isAllPaused. When isAllPaused will be true, the array will get filled by true values and vice versa, after this we can handle individual pause and resume too.

File: src/components/Connection.tsx [hunk 5]
```diff
@@ -50,6 +50,10 @@ interface GridViewProps {
   isGridView: boolean;
   setIsGridView: React.Dispatch<React.SetStateAction<boolean>>;
 }
+interface DisplayProps {
+  isDisplay: boolean;
+  setIsDisplay: React.Dispatch<React.SetStateAction<boolean>>;
+}
```
Comment (akadeepesh) [inline src/components/Connection.tsx]: We don't need to define another interface if we are adding isDisplay and it's set variable in ConnectionProps, one interface is enough.

File: src/components/Connection.tsx [hunk 6]
```diff
@@ -88,6 +96,10 @@ const Connection: React.FC<ConnectionProps> = ({
   const readerRef = useRef<
     ReadableStreamDefaultReader<Uint8Array> | null | undefined
   >(null); // Ref to store the reader for the serial port
+  const [isPrimary, setIsPrimary] = useState(false);
+
+  // Condition to change the variant
+  const variant = isPrimary ? "primary" : "outline";
```
Comment (akadeepesh) [inline src/components/Connection.tsx]: This is of no use, remove this, if error occurs, use ternary operator specifically there.

**Review by akadeepesh — CHANGES_REQUESTED** [review]

**Comment by akadeepesh** [comment]
@Ritika8081  After fixing these please mark "resolve conversation" and tag me to get this PR merged.

File: src/components/Connection.tsx (hunk 5 above)
Comment (Ritika8081) [reply src/components/Connection.tsx]: okay I'll remove it.

File: src/components/Canvas.tsx (hunk 4 above)
Comment (Ritika8081) [reply src/components/Canvas.tsx:34]: I have updated code now I my approach is to iterates over the entire chartRef.current array and applies the same action (start or stop) to every chart, regardless of their individual state.

File: src/components/Canvas.tsx (hunk 2 above)
Comment (Ritika8081) [reply src/components/Canvas.tsx:36]: okay I'll take care of this.

File: src/components/Connection.tsx (hunk 6 above)
Comment (Ritika8081) [reply src/components/Connection.tsx]: okay

**Pull request merged**